plugin-server/src/worker/ingestion/app-metrics.ts

```diff
@@ -0,0 +1,124 @@
+import { Message } from 'kafkajs'
+import { DateTime } from 'luxon'
+
+import { KAFKA_APP_METRICS } from '../../config/kafka-topics'
+import { Hub, TeamId, TimestampFormat } from '../../types'
+import { status } from '../../utils/status'
+import { castTimestampOrNow } from '../../utils/utils'
+
+export interface AppMetricIdentifier {
+    teamId: TeamId
+    pluginConfigId: number
+    jobId?: string
+    category: 'processEvent' | 'onEvent' | 'exportEvents'
+}
+
+export interface AppMetric extends AppMetricIdentifier {
+    successes?: number
+    successesOnRetry?: number
+    failures?: number
+}
```

Comment:
even if we log it this way to CH in the end (which I'm not sure if we should, see my comment in the sql file), then here I propose we have it like this providing us more flexibility and it would make the call locations simpler too:
```suggestion
export interface AppMetric extends AppMetricIdentifier {
    successes?: number
    failures?: number
    attempt: number
}
```

Reply:
I'm intentionally keeping this identical to schema-in-database. Less data transformations is better.

Let's keep the schema discussion under a single thread: https://github.com/PostHog/posthog/pull/12143#discussion_r991701129